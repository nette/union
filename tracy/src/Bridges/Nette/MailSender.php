<?php

/**
 * This file is part of the Tracy (https://tracy.nette.org)
 * Copyright (c) 2004 David Grudl (https://davidgrudl.com)
 */

namespace Tracy\Bridges\Nette;

use Nette;
use Tracy\Dumper;
use Tracy\Helpers;


/**
 * Tracy logger bridge for Nette Mail.
 */
class MailSender
{
	use Nette\SmartObject;

	/** @var Nette\Mail\IMailer */
	private $mailer;

	/** @var string|null sender of email notifications */
	private $fromEmail;


	public function __construct(Nette\Mail\IMailer $mailer, $fromEmail = null)
	{
		$this->mailer = $mailer;
		$this->fromEmail = $fromEmail;
	}


	/**
	 * @param  string|\Exception|\Throwable  $message
	 * @param  string  $email
	 * @return void
	 */
	public function send($message, $email)
	{
		$host = preg_replace('#[^\w.-]+#', '', isset($_SERVER['HTTP_HOST']) ? $_SERVER['HTTP_HOST'] : php_uname('n'));

		$mail = new Nette\Mail\Message;
		$mail->setHeader('X-Mailer', 'Tracy');
		$mail->setFrom($this->fromEmail ?: "noreply@$host");
		$mail->addTo($email);
		$mail->setSubject('PHP: An error occurred on the server ' . $host);
		$mail->setBody(static::formatMessage($message) . "\n\nsource: " . Helpers::getSource());

		$this->mailer->send($mail);
	}


	/**
	 * @param  mixed  $message
	 * @return string
	 */
	private static function formatMessage($message)
	{
		if ($message instanceof \Exception || $message instanceof \Throwable) {
			while ($message) {
				$tmp[] = ($message instanceof \ErrorException
					? Helpers::errorTypeToString($message->getSeverity()) . ': ' . $message->getMessage()
					: Helpers::getClass($message) . ': ' . $message->getMessage() . ($message->getCode() ? ' #' . $message->getCode() : '')
				) . ' in ' . $message->getFile() . ':' . $message->getLine();
				$message = $message->getPrevious();
			}
			$message = implode("\ncaused by ", $tmp);

		} elseif (!is_string($message)) {
			$message = Dumper::toText($message);
		}

		return trim($message);
	}
}
