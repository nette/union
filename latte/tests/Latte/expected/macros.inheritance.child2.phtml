<?php
%A%

class Template%a% extends Latte\Template
{
	public $blocks = [
		'content' => 'blockContent',
		'title' => 'blockTitle',
		'sidebar' => 'blockSidebar',
	];

	public $blockTypes = [
		'content' => 'html',
		'title' => 'html',
		'sidebar' => 'html',
	];


	function render()
	{
%A%
<?php
		call_user_func(reset($this->blockQueue['content']), get_defined_vars());
		?>

<?php
		call_user_func(reset($this->blockQueue['sidebar']), get_defined_vars());
		?>	<?php
	}


	function prepare()
	{
		extract($this->params);
		$this->parentName = "parent";
		return get_defined_vars();
	}


	function blockContent($_args)
	{
		extract($_args);
		?>	<h1><?php
		call_user_func(reset($this->blockQueue['title']), get_defined_vars());
		?></h1>

	<ul>
<?php
		$iterations = 0;
		%a%
		foreach ($people as $person) {
			?>		<li><?php echo LFilters::escapeHtml($person) ?></li>
<?php
			$iterations++;
		}
		?>
	</ul>
<?php
	}


	function blockTitle($_args)
	{
		?>Homepage <?php
	}


	function blockSidebar($_args)
	{

	}

}
