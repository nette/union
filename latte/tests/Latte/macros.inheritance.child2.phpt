<?php

/**
 * Test: Nette\Latte\Engine: {extends ...} test II.
 *
 * @author     David Grudl
 */

use Nette\Latte,
	Tester\Assert;


require __DIR__ . '/../bootstrap.php';


$latte = new Latte\Engine;

Assert::matchFile(
	__DIR__ . '/expected/macros.inheritance.child2.phtml',
	$latte->compile(__DIR__ . '/templates/inheritance.child2.latte')
);
Assert::matchFile(
	__DIR__ . '/expected/macros.inheritance.child2.html',
	$latte->renderToString(
		__DIR__ . '/templates/inheritance.child2.latte',
		array('people' => array('John', 'Mary', 'Paul'))
	)
);
