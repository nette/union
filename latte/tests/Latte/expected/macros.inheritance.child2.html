<!DOCTYPE html>
<head>
	<title>Homepage </title>
</head>

<body>
	<div id="sidebar">
	</div>

	<div id="content">
	<h1>Homepage </h1>

	<ul>
		<li>John</li>
		<li>Mary</li>
		<li>Paul</li>
	</ul>
	</div>
</body>
</html>
Parent: main/extends
